.github/workflows/check-go-task.yml

```diff
@@ -219,6 +219,55 @@ jobs:
       - name: Check whether any tidying was needed
         run: git diff --color --exit-code
 
+  check-license-headers:
+    name: check-license-headers (${{ matrix.module.path }})
+    needs: run-determination
+    if: needs.run-determination.outputs.result == 'true'
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+
+      matrix:
+        module:
+          - path: ./
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Install Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Check license headers
+        env:
+          GO_MODULE_PATH: ${{ matrix.module.path }}
+        run: |
+          # We read the template content and we modify the variadic date to be always 2000
+          LICENSE=$(cat ./license_header.tpl | sed "3s|.*|Copyright 2000 ARDUINO SA (http://www.arduino.cc/)|")
+          N_LINES=$(wc -l < ./license_header.tpl)
+          STATUS=0
+          for file in $(
+            find . \
+              -type d -name 'testdata' -prune -o \
+              -type d -name 'legacy' -prune -o \
+              -type f ! -name '*_grpc.pb.go' \
+              -type f ! -name 'json_easyjson.go' \
+              -type f ! -name 'index_easyjson.go' \
+              -regex ".*\.go$" -print
+          ); do
+          # We read only the license header, removing all the comments symbols and changing the date to be 2000. This will
+          # help us understand if the other part of the license are exactly the same as the template.
+          HEADER=$(head -n$N_LINES $file | sed -r 's| [0-9]{4} | 2000 |' | sed 's|// ||; s|^//||' )
```

Comment:
We must keep the year range `2020-2023`. I did some research and found that:

1. the starting year must be kept to indicate when the copyright started (otherwise any previous work may not be considered under copyright)
2. the ending year must be kept to indicate when the copyright expires (it happens after the last year + some decades actually...)
3. we cannot use `2020-present`...
4. old headers should be updated every time a file is touched (for example if a file made in 2020 is modified in 2023 then the ending year in the header must be updated with `2020-2023`)

At this point, maybe the simplest solution is to just put the year range in the template file and update the template  + all source files every year?